up42/order.py

```diff
@@ -146,18 +155,38 @@ def track_status(self, report_time: int = 120) -> str:
         Returns:
             str: The final order status.
         """
+
+        def substatus_messages(substatus: str) -> str:
+            substatus_user_messages = {
+                "FEASIBILITY_WAITING_UPLOAD": "feasibility study in progress, wait for updates.",
+                "FEASIBILITY_WAITING_RESPONSE": "choose a feasibility option",
+                "QUOTATION_WAITING_UPLOAD": "quotation being prepared, wait for updates.",
+                "QUOTATION_WAITING_RESPONSE": "review to accept or reject price",
+                "QUOTATION_ACCEPTED": "wait for files to be delivered.",
+            }
+            if substatus in substatus_user_messages:
+                message = substatus_user_messages[substatus]
+                return f"{substatus}, {message}"
+            return f"{substatus}"
+
         logger.info(
-            f"Tracking order status, reporting every {report_time} seconds...",
+            f"Tracking order status, reporting every {str(report_time)} seconds...",
         )
         time_asleep = 0
 
+        # check order details and react for tasking orders.
+
         while not self.is_fulfilled:
             status = self.status
             if status in ["PLACED", "BEING_FULFILLED"]:
                 if time_asleep != 0 and time_asleep % report_time == 0:
                     logger.info(f"Order is {status}! - {self.order_id}")
+                if self.info["type"] == "TASKING":
+                    logger.info(substatus_messages(self.order_details["subStatus"]))
```

Comment:
```suggestion
                    logger.info(substatus_messages(self.order_details.get("subStatus")))
```

Consider using `.get` method so we don't throw KeyError in case tasking ever updates this keyword.